p4-16/spec/P4-16-spec.mdk

```diff
@@ -6639,6 +6639,39 @@ table is an abstract finite map that is given a key and produces as a
 result either an action or a "miss" indication, as described in
 Section [#sec-mau-semantics].
 
+The expected meaning of these values is as follows:
+
+- an `exact` match kind on a key field means that the value of the
+  field in the table specifies exactly all the bits for the key field.
+  This is applicable for key fields of all types supported by the target
+  which have an equality operation.
+
+- a `ternary` match kind on a key field means that the field in the
+  table specifies a set of values for the key field using a value, a
+  mask, and a priority.  The meaning of the `(value, mask)` pair is
```

Comment:
Comment on "and a priority".  One might get the impression from reading this that if the table has 3 ternary fields, then there are 3 priorities.  There are not.  There is only one priority value for the entire table entry.

One way to avoid causing such confusion would be not to mention priorities except in relation to an entire table entry, later.  When describing the match condition for a single field, omit any mention of priority.

Thus this bullet item could be replaced with:
```
- a `ternary` match kind on a key field means that the field in the
  table specifies a set of values for the key field using a value and
  a mask.  The meaning of the `(value, mask)` pair is
  similar to the P4 mask expressions, as described in Section [#sec-cubes]:
  a key field `k` matches the table entry when `k & mask == value & mask`.
```

Then later after all of the bullet points that describe exact, ternary, and lpm, only THEN bring up the issue that some table entries require a priority (those with at least one `ternary` field), and some table entries do not require a priority (those with no `ternary` fields).  The priority is not used to determine which entries match, only which of multiple matching entries "wins", and its action is executed.

Reply:
Updated text in commit 2 resolves this comment.